### ISSUE TITLE
new tag v.1.30.5 - incompatibilities

### ISSUE THREAD

**Issue by oblomov-dev** [abap2UI5/abap2UI5#646]
Created a new tag [v.1.30.5](https://github.com/abap2UI5/abap2UI5/releases/tag/v.1.30.5)

FYI @choper725
incompatible renaming of nest_view_display2 to nest2_view_display, to make the names of all display methods more consistent.

**Comment by choper725** [comment]
something changed in the factory of xml view ??

**Comment by oblomov-dev** [comment]
No small improvements but they should not effect anything outside (I hope).

**Comment by oblomov-dev** [comment]
Cleaned up the framework and unfortunately added some more small incompatibilities with https://github.com/abap2UI5/abap2UI5/pull/650

(1) the method client-title_set is replaced by a custom control. Check the example here:
https://github.com/abap2UI5/abap2UI5-samples/blob/bcc42de145ac01c24e52f0511ce4b969139fb661/src/z2ui5_cl_demo_app_125.clas.abap#L39

(2) the method client-timer_set is replaced by a custom control (it is loaded per default with the start of abap2UI5) and can be used as follows:
https://github.com/abap2UI5/abap2UI5-samples/blob/bcc42de145ac01c24e52f0511ce4b969139fb661/src/z2ui5_cl_demo_app_028.clas.abap#L97
Or to load custom controls here:
https://github.com/abap2UI5/abap2UI5-samples/blob/bcc42de145ac01c24e52f0511ce4b969139fb661/src/z2ui5_cl_demo_app_075.clas.abap#L80

(3) the helper methods (get_url_source_code, app_get_url, url_param_get, url_param_set) of the class z2ui5_cl_xml_view are not a part of abap2UI5 anymore, but if needed can be copy&pasted from here:
https://github.com/abap2UI5/abap2UI5-samples/blob/bcc42de145ac01c24e52f0511ce4b969139fb661/src/z2ui5_cl_demo_utility.clas.abap#L14

I have in mind to add more functionalities in the future with custom controls. Custom Controls can be used on app level only and also just when it is needed which keeps the complexity of the abap2UI5 itself as small as possible. Hope this makes extensions and maintenance in the future easier. But as always, we will see if this will work out and let me know if you run into problems.

**Comment by choper725** [comment]
hi @oblomov-dev ,
great that you think about maintanence in mind for the progress of this prooject,

one thing i think the removal of set title and timer is making without any replacment for the current use (and correct me if im wrong), is the the timer set and title set before the change can be used without z2ui5 xml object, which is in my view remove very dynamic option, an example will be is to make a page which is refreshing the data model every 1 minute,
 now with this change is not possible anymore (i think), now we must rerender the whole view, if i get this right.

in conclusion in think that the timer and title somewhat independent of the xml view and offer an extra layer of capebilities once they are not tied to it

**Comment by oblomov-dev** [comment]
yes understand your point, want to make it definitely usable without being forced to rerender the whole view.

**timer:**
Therefore added also the property "checkrepeat" for the timer:
https://github.com/abap2UI5/abap2UI5-samples/blob/4a8c55580b91af4c44075c85814ce4f7c02a3a19/src/z2ui5_cl_demo_app_028.clas.abap#L100

Additionally there would be also the option to make it changeable only with view_update_model() but did not check this further but when there is the need I will try to add this.

background:
Using the timer in a cc makes it very easy to trigger events, unfortunately I had a lot of trouble (and redundant functionality) before here:
https://github.com/abap2UI5/abap2UI5/blob/33f1101f2b5e92af85f193e59fa29ddd053c6b08/src/z2ui5_cl_fw_http_handler.clas.abap#L196

So i hope we can catch every use case with the CC approach, so feel free to share the example with a use case which is not working with the new approach that i can adjust it.

**browser title:**
I had in mind that it will be changed only once per view but if you need the fucntionality to do it more often without rerendering i can also try to add this.

**Comment by choper725** [comment]
well i looked into it and made small example,
it seems to work once insetred once to the view with check repeat but seems that model updating is not happening once a popup is open, (still the timer is running though)...
check open popup.


with open popover,
the timer closes the popover once it is running, 
what was not happening using the client timer as far i checked..

here it is..

tbh, it does seems more convenient this way..

[timer popup popover demo.txt](https://github.com/abap2UI5/abap2UI5/files/13390897/timer.popup.popover.demo.txt)

**Comment by oblomov-dev** [comment]
yes i got the idea from the pull to refresh functionality:
https://sapui5.hana.ondemand.com/sdk/#/entity/sap.m.PullToRefresh

Maybe UI5ers don't like to put controller (and non UI) logic into a view but it fits perfect to abap2UI5 because everything is sent via XML-View from the backend anyway. More functionality can be added easily with this i hope.

Back to the example:
(1) I added now the old timer again, so at the moment both can be used. Maybe this helps to bring the old functionality to the new approach first, before deleting it and create incompatibilites.
(2) Added your example to the samples:
https://github.com/abap2UI5/abap2UI5-samples/blob/main/src/z2ui5_cl_demo_app_129.clas.abap
(3) Activated the old timer again, but the popover also gets closed after a roundtrip.

Can you adjust the example with the behaviour you need and send a PR?

**Comment by choper725** [comment]
well i checked with the old timer..
and seems the popover do close on a roundtrip.. i wonder why, as the view is not being changed...

and about the popup is behaves exactly like in the new timer method.. the timer is running but the model dont get really updated once the popup is on the screen,

seems like the new timer is exactly the same as the old one.. just less flexible in terms of where you can call it in the abap program..

**Comment by oblomov-dev** [comment]
every roundtrip displays a busy indicator (sometimes it is just too fast to see it), maybe this leads to the closing of the popover. If this functionality is needed i can check if deactivating the busy indicator is an option, maybe this helps keeping the popover visible.

yes it is less flexible in the application, but a lot more simple for the abap2UI5 logic. i will keep both timers in the framework for the next weeks and you have time to check if you run into more problems you can not solve with the new timer and feel free to try to change my mind.

**Comment by choper725** [comment]
well i checked all the usecases for timer that i use and it seems that its behave exactly the same, so if you i intend to remove the client version now is better than later :)

**Comment by oblomov-dev** [comment]
ok great, thank you for checking this!

i deleted now the timer method of the interface again. but i will keep the logic in the framework for a while, so i can easily activate it again, if errors show up or someone else runs into problems.

**Comment by choper725** [comment]
its not the busyindicator its the destroy of popover once view = MAIN
https://github.com/abap2UI5/abap2UI5/blob/7629b5012126a4f1de997576fffdf985e9e20fe0/src/z2ui5_cl_fw_http_handler.clas.abap#L348

**Comment by oblomov-dev** [comment]
ah good to know, maybe we can delete the popover destroy line, i will check this again.

ok so i hoped you can decide which model is updated via the specific method in the backend view_model_upadte, popover_model_update etc.
But this also worked before or do we need a new functionality here?

**Comment by choper725** [comment]
its just strange that updating the main model while popup is open does nothing…
or maybe its not and thats just the ui5 way..
anyway i dont think its needed or time needs to be spend on this..

**Comment by oblomov-dev** [comment]
yes all the model logic and when something gets updated and when not etc. is also a bit magic to me. we create different json models based on the same data with these main, popup, nest etc. I also discover new behaviours and functions of the ui5 frameowrk while working on this project all the time. as far as all we need works for now, we keep it like this for the moment. just open a new issue when you need more and i will close this one.

**Comment by WegnerDan** [comment]
> (1) the method client-title_set is replaced by a custom control. Check the example here: [abap2UI5/abap2UI5-samples@`bcc42de`/src/z2ui5_cl_demo_app_125.clas.abap#L39](https://github.com/abap2UI5/abap2UI5-samples/blob/bcc42de145ac01c24e52f0511ce4b969139fb661/src/z2ui5_cl_demo_app_125.clas.abap#L39)

Is it strictly necessary to use the timer custom control just to set the title?

**Comment by choper725** [comment]
well you got a point there..

**Comment by oblomov-dev** [comment]
yes true, true.
just technically there is the need to load the custom control with the timer control, but we can also load the most needed or custom controls which basic logic (like title) per default at startup. i did this with the timer control here:
https://github.com/abap2UI5/abap2UI5/blob/774bcde0192661c4d03435a06b7813b58ab80be5/src/z2ui5_cl_fw_http_handler.clas.abap#L554

Maybe we add title here and some more? And additionally giving the option to have an importing parameter to influence which CC are loaded per default at startup.

**Comment by choper725** [comment]
well i dont think the issue is the timer.
maybe because the title its just 1 liner in the http_handler, its not that bad to leave it there with the client->title_set

**Comment by oblomov-dev** [comment]
yes of course we can discuss if a custom control for the one liner title is the best, but i want to make it more consistent because i also put the scroll and focus function in a separated CC, so the title should also stay there. All browser functions are now separated from the index.html with this. But yes understand, when this is not working for you or makes too much problems, let me know that we can think about changing this again.

Added the option to define which CC should be loaded at startup:
https://github.com/abap2UI5/abap2UI5/blob/abb818141db639fde102de04363b80a3253aa3ea/src/z2ui5_cl_fw_http_handler.clas.abap#L20

Separated CC loading with the timer function on app level is not needed in this situation, adjusted the title demo:
https://github.com/abap2UI5/abap2UI5-samples/blob/main/src/z2ui5_cl_demo_app_125.clas.abap

Per default the following CC are loaded:
https://github.com/abap2UI5/abap2UI5/blob/abb818141db639fde102de04363b80a3253aa3ea/src/z2ui5_cl_fw_http_handler.clas.abap#L49

Give it a try and let me know how this works for you! 

Everything is experimental and i am open for changes, just try to find good ways to keep the complexity of all this functionality under control because the project is constantly getting bigger. I also need to add the new interface to all CC classes, but first want to test this more.

**Comment by oblomov-dev** [comment]
fyi: updated the importing of custom js again, see the edited comment from before. Overlooked that there was already an importing parameter for custom_js.

**Comment by WegnerDan** [comment]
This new version works well for me!

**Comment by WegnerDan** [comment]
Z2UI5_CL_DEMO_APP_084 seems to be broken currently.

`Uncaught (in promise) ModuleError: failed to load 'sap/ui/webc/main/Title.js' from https://ui5.sap.com/1.96.21/resources/sap/ui/webc/main/Title.js: 404`

![image](https://github.com/abap2UI5/abap2UI5/assets/6908247/802bda47-4f39-4c6d-bdca-1831bea1ae3f)

**Comment by choper725** [comment]
this demo is not using the new messages cc, see demo 135 for that

**Comment by WegnerDan** [comment]
I'm having an unrelated issue with message_view in a popover and wanted to see if this works in the demo where it's used. 
84 seems to be broken and it looks like it has to do with the custom controls.

**Comment by oblomov-dev** [comment]
For Message Views in Popovers check:
Z2UI5_CL_DEMO_APP_038

And for Messaging/Message Manager the new demo is:
z2ui5_cl_demo_app_135

Fixed demo 84 to see Constraints and Format Options.

Updated the samples repository with https://github.com/abap2UI5/abap2UI5-samples/commit/251a82d2633ccdc55bd6d45124e5c33b5580ca12

**Comment by oblomov-dev** [comment]
FYI

Updated the title CC and the demo z2ui5_cl_demo_app_125.

Now the binding can be used with data binding and behaves similar to all other UI5 controls. For example the title can now be changed directly at the frontend without server roundtrip.

**Comment by oblomov-dev** [comment]
continue here https://github.com/abap2UI5/abap2UI5/issues/676
